cfgrammar/src/lib/yacc/parser.rs

```diff
@@ -104,6 +104,7 @@ lazy_static! {
     static ref RE_NAME: Regex = { Regex::new(r"^[a-zA-Z_.][a-zA-Z0-9_.]*").unwrap() };
     static ref RE_TOKEN: Regex =
         { Regex::new("^(?:(\".+?\")|('.+?')|([a-zA-Z_][a-zA-Z_0-9]*))").unwrap() };
+    static ref RE_ACTION: Regex = { Regex::new(r"^\{[^\{]*\}").unwrap() };
```

Comment:
We can't use a regex for actions, because regexs can't count brackets. For example this is a valid action block:

```rust
{
   if ... { } else { }
}
```
but the regex will (I think) only match `{ if ... { }`.

I think we're going to have to write a little hand-coded action matcher which counts open/close curlies so that it knows when to stop. It's probably only 5 or 6 LoC.

Reply:
Whoops, you are right. For some reason I didn't think about having to match curly braces within the actions.